tea/scenes/apply.go

```diff
@@ -0,0 +1,226 @@
+		modLog := fmt.Sprintf("Installing mods from profile %s\n", profile.Name)
+
+		for _, mod := range profile.Mods {
+			modLog += fmt.Sprintf("%s (%s): %s\n", mod.Name, mod.ID, mod.Version)
+			model.log <- modLog
+
+			versions, err := ficsit.ModVersions(context.TODO(), root.GetAPIClient(), mod.ID, ficsit.VersionFilter{
+				Limit:    100,
+				Offset:   0,
+				Search:   mod.Version,
+				Order:    ficsit.OrderDesc,
+				Order_by: ficsit.VersionFieldsCreatedAt,
+			})
+
+			if err != nil {
+				panic(err) // TODO
+			}
+
+			modCount := len(versions.GetMod.Versions)
+			if modCount == 0 {
+				modLog += "No versions found"
+				continue
+			}
+
+			version := versions.GetMod.Versions[0]
+			url := fmt.Sprintf("https://api.ficsit.app/v1/mod/%s/versions/%s/download", mod.ID, version.Id)
+			modPath := path.Join(installation, mod.Reference)
+			modZipPath := modPath + ".smod"
+
+			modLog += fmt.Sprintf("Downloading %s %s to %s\n", mod.Name, version.Version, modZipPath)
+			model.log <- modLog
+
+			// Get the data
+			resp, err := http.Get(url)
+			if err != nil {
+				modLog += fmt.Sprintf("Could not download: %s\n", err)
+				model.log <- modLog
+				continue
+			}
+			defer resp.Body.Close()
+
+			// Create the file
+			out, err := os.Create(modZipPath)
+			if err != nil {
+				modLog += fmt.Sprintf("Could not create output file %s:]n %s\n", modZipPath, err)
+				model.log <- modLog
+				continue
+			}
+			defer out.Close()
+
+			// Write the body to file
+			_, err = io.Copy(out, resp.Body)
+			if err != nil {
+				modLog += fmt.Sprintf("Failed to write output file %s:\n %s\n", modZipPath, err)
+				model.log <- modLog
+				continue
+			}
+
+			modLog += "Download complete\n"
+			model.log <- modLog
+
+			err = os.MkdirAll(modPath, 0775)
+			if err != nil {
+				modLog += fmt.Sprintf("Failed to create mod folder %s:\n %s\n", modPath, err)
+				model.log <- modLog
+				continue
+			}
+
+			// Open a zip archive for reading.
+			r, err := zip.OpenReader(modZipPath)
+			if err != nil {
+				modLog += fmt.Sprintf("Failed to open %s:\n %s\n", modZipPath, err)
+				model.log <- modLog
+				continue
+			}
+			defer r.Close()
+
+			for _, f := range r.File {
+				modLog += fmt.Sprintf("Unzipping %s\n", f.Name)
+				pluginFile, err := f.Open()
+				if err != nil {
+					modLog += fmt.Sprintf("Unable to open %s in %s:\n %s\n", f.Name, modZipPath, err)
+				}
+
+				destPath := path.Join(modPath, f.Name)
+				destDir := path.Dir(destPath)
+				err = os.MkdirAll(destDir, 0775)
+				if err != nil {
+					modLog += fmt.Sprintf("Failed to create mod subfolder %s:\n %s\n", modPath, err)
+					model.log <- modLog
+					continue
+				}
+	
+				out, err := os.Create(destPath)
+				if err != nil {
+					modLog += fmt.Sprintf("Could not create output file %s:\n %s\n\n", destPath, err)
+					model.log <- modLog
+					continue
+				}
+	
+				io.Copy(out, pluginFile)
+			}
+			modLog += "Done"
+			model.log <- modLog
+
+		}
+	}()
```

Comment:
This should not be handled in the TUI. `ficsit-cli` will be further used as a library for SMM to have consistent mod management, as well as a CLI-only mode for `ficsit-cli`, so please follow separation of concerns.

This is also missing any kind of dependency resolution or fetching. The download link should also not be crafted, but taken from the API response, as it may change over time.

This is also not utilizing any concept of lockfiles, and just logs any errors without retry or proper failure management.

A cache should also be utilized so the end-user doesn't have to re-download the same mods several times.

The system also needs to be backwards-compatible with current SMM implementation and do a migration over to the new system.